tests/lib/nested.sh

```diff
@@ -0,0 +1,210 @@
+#!/bin/bash
+
+WORK_DIR="${WORK_DIR:-/tmp/work-dir}"
+SSH_PORT=${SSH_PORT:-8022}
+MON_PORT=${MON_PORT:-8888}
+IMAGE_FILE="${WORK_DIR}/ubuntu-core-20.img"
+
+execute_remote(){
+    sshpass -p ubuntu ssh -p "$SSH_PORT" -o ConnectTimeout=10 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no ubuntu@localhost "$*"
+}
+
+wait_for_service() {
+    local service_name="$1"
+    local state="${2:-active}"
+    for i in $(seq 300); do
+        if systemctl show -p ActiveState "$service_name" | grep -q "ActiveState=$state"; then
```

Comment:
`systemctl is-active`?

Reply:
I assume it's to support multiple different states, and not just active - however it's not something that is required for this instance, so I think we might just go with your suggestions here